### PR TITLE
Humble

### PR DESCRIPTION
**Pull request by Ryujiyasu**
### check and fix tutorials for working on Humble

- nav2_costmap_filters_demo->WORKING
- nav2_gradient_costmap_plugin->WORKING
- nav2_straightline_planner->WORKING
- nav2_sms_behavior->**NOT WORKING**
- sam_bot_description->**NOT WORKING**

### fix point

**nav2_sms_behavior**

- change "recovery" -> "behavior"

**sam_bot_description**

- add "xacro" to CMakeLists.txt

thank you

### CODE REVIEW

File: nav2_sms_behavior/CMakeLists.txt [hunk 1]
```diff
@@ -19,8 +19,11 @@ find_package(builtin_interfaces REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(nav2_costmap_2d REQUIRED)
 find_package(nav2_core REQUIRED)
-find_package(nav2_recoveries REQUIRED)
+find_package(nav2_behaviors REQUIRED)
 find_package(pluginlib REQUIRED)
+find_package(rosidl_default_generators REQUIRED)
+find_package(action_msgs REQUIRED)
+find_package(CURL REQUIRED)
```
Comment (SteveMacenski) [inline nav2_sms_behavior/CMakeLists.txt:26]: Why CURL?

File: nav2_sms_behavior/package.xml [hunk 2]
```diff
@@ -25,15 +25,17 @@
   <depend>builtin_interfaces</depend>
   <depend>tf2_ros</depend>
   <depend>nav2_costmap_2d</depend>
-  <depend>nav2_recoveries</depend>
+  <depend>nav2_behaviors</depend>
   <depend>nav2_core</depend>
   <depend>pluginlib</depend>
+  <depend>action_msgs</depend>
+  <depend>libcurl</depend>
```
Comment (SteveMacenski) [inline nav2_sms_behavior/package.xml:32]: This doesn't seem like the right rosdep key https://github.com/ros/rosdistro/blob/4f287cd19d8c3127b444ecfca0a0264569aa71de/rosdep/base.yaml#L691

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

**Comment by SteveMacenski** [comment]
Thanks! Much appreciated!

File: nav2_sms_behavior/CMakeLists.txt (hunk 1 above)
Comment (Ryujiyasu) [reply nav2_sms_behavior/CMakeLists.txt:26]: use curl in  Twillio API(twillio.cpp)
if i don't use package curl , failed to send sms.

File: nav2_sms_behavior/package.xml (hunk 2 above)
Comment (Ryujiyasu) [reply nav2_sms_behavior/package.xml:32]: Can you give me some advice on how to change it?

File: nav2_sms_behavior/CMakeLists.txt (hunk 1 above)
Comment (Ryujiyasu) [reply nav2_sms_behavior/CMakeLists.txt:26]: `CURL`
->
`curl`

File: nav2_sms_behavior/package.xml (hunk 2 above)
Comment (SteveMacenski) [reply nav2_sms_behavior/package.xml:32]: Correct

File: nav2_sms_behavior/CMakeLists.txt (hunk 1 above)
Comment (Ryujiyasu) [reply nav2_sms_behavior/CMakeLists.txt:26]: how can i change package CURL?

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**